diff -ru KL_FRIENDMOD_V4-io2/lua/FRIENDMOD.lua KL_FRIENDMOD_V4-io2_strashEdit.r002/lua/FRIENDMOD.lua
--- KL_FRIENDMOD_V4-io2/lua/FRIENDMOD.lua	2021-08-21 10:07:59.625684324 +0200
+++ KL_FRIENDMOD_V4-io2_strashEdit.r002/lua/FRIENDMOD.lua	2021-08-21 10:09:56.024086735 +0200
@@ -420,6 +420,9 @@
 local savedtheworld = nil -- should we do special celebrations for 1 player beating a boss, or leader bonus deciding a normal teams game?
 local restorecolors = true -- have we written colors back to teams-mode players during intermission yet? (gotta preserve personal colors for mixed games)
 
+-- [strashEdit]
+local doHUDstuff= true
+
 -- just in case.
 local function sanitizebossname(s)
 	return s:gsub("%W", "")
@@ -976,7 +979,13 @@
 	return true
 end
 
-local function allfinished()
+local function allfinished()
+	-- [strashEdit] --
+	-- trying this out… hopefully it'll make it work properly with Elim
+	if ((Elimination and Elimination()) or (hpmod and hpmod.running)) then
+		return orangefinished() or bluefinished()
+	end
+	--
 	for p in players.iterate
 		if not finished(p) then
 			return false
@@ -1474,8 +1483,12 @@
 	end
 	if leveltime == 5 then
 		if bossmode and doteamstuff then
-			S_StartSound(nil, sfx_frboss)
-		else
+			S_StartSound(nil, sfx_frboss)
+		-- [strashEdit]
+		--  strashbot server will handle those
+		--else
+		elseif not strash_mods then
+		-- --
 			S_StartSound(nil, sfx_fmssb0)
 			if doteamstuff then
 				S_StartSound(nil, sfx_fmteam)
@@ -1721,7 +1734,14 @@
 				if not supervalid(p) then
 					continue
 				end
-				if p.FRteam == 2 and not (p.pflags & PF_TIMEOVER) then
+				if p.FRteam == 2 and (not (p.pflags & PF_TIMEOVER))
+					and (not (Elimination and Elimination()))
+					-- [strashEdit] --
+					-- yeah, blowing up and therefore losing when all other team
+					-- is dead is also problematic in HPmod
+					and (not (hpmod and hpmod.running))
+					--  --
+				then
 					FRlog(p, "blue blowout die")
 					welcometofuckingdie(p)
 				end
@@ -1734,7 +1754,14 @@
 				if not supervalid(p) then
 					continue
 				end
-				if p.FRteam == 1 and not (p.pflags & PF_TIMEOVER) then
+				if p.FRteam == 1 and (not (p.pflags & PF_TIMEOVER))
+					and (not (Elimination and Elimination()))
+					-- [strashEdit] --
+					-- yeah, blowing up and therefore losing when all other team
+					-- is dead is also problematic in HPmod
+					and (not (hpmod and hpmod.running))
+					--  --
+				then
 					FRlog(p, "orange blowout die")
 					welcometofuckingdie(p)
 				end
@@ -1854,6 +1881,7 @@
 			FRlog(p, "doing allfinish iterator")
 			if p.exiting == 99 and not p.FRstasis then
 				FRlog(p, "Detonate handler")
+				print("Detonate handler "..p.name)
 				p.FRstasis = 5 * TICRATE - 2
 			end
 			if p.FRstasis == nil then
@@ -1912,10 +1940,19 @@
 
 hud.add(function(v, p, c) -- AAAAAAAAAAHHHHHHHHHHHHH
 	if G_BattleGametype() return end -- nah
-	localplayer = p
-	if (countactive() < 2 and not fr_debug.value) then
+	localplayer = p
+	-- [strashEdit] --
+	-- yeah these modes might still need friend's hud if only 1 player remains…
+	if (not ((Elimination and Elimination()) or (hpmod and hpmod.running)))
+		and (countactive() < 2 and not fr_debug.value)
+	then
 		return
-	end
+	end
+	-- --
+
+	-- [strashEdit]
+	if not doHUDstuff return end
+	
 	if not doteamstuff
 		if leveltime < (6 * TICRATE + (3 * TICRATE / 4)) then
 			local ffaflag = 0
@@ -2106,11 +2143,15 @@
 		elseif not p.spectator and p.FRexiting != 1 -- shhh. no tears.
 			local sortedplayers = {}
 			for q in players.iterate
-				if q.mo and not q.spectator and q.FRready and not (q.pflags & PF_TIMEOVER) and q.realtime <= bossplayer.realtime and q.FRteam == 1 then
+				-- if q.mo and not q.spectator and q.FRready and not (q.pflags & PF_TIMEOVER) and q.realtime <= bossplayer.realtime and q.FRteam == 1 then
+				if q.mo and not q.spectator and q.FRready then
 					table.insert(sortedplayers, q)
 				end
 			end
 			table.sort(sortedplayers, function(a, b)
+				if(a.FRteam~=b.FRteam) then
+					return a.FRteam==1
+				end
 				return a.realtime < b.realtime
 			end)
 			for _, q in ipairs(sortedplayers)
@@ -2251,4 +2292,38 @@
 
 rawset(_G, "FRIENDMOD_CheckTeams", function(...)
 	return fr_enabled.value
-end)
\ No newline at end of file
+end)
+
+
+--- [strashEdit] ---
+--- allowing external script to check for scores
+--- and for players team belonging
+
+-- used originaly by NRJV's elimination and therefor the
+-- strashbot's tweaked version
+rawset(_G, "FRIENDMOD_TeamBattle", function(...)
+	return doteamstuff
+end)
+
+rawset(_G, "FRIENDMOD_GetScores", function(...)
+	return finalrecalc, bluescore, orangescore, bossplayer, savedtheworld, blueprefix() ,orangeprefix()
+end)
+
+rawset(_G, "FRIENDMOD_GetPlayersData", function(p)
+	return {
+		FRteam= p.FRteam,
+		personalscore= p.personalscore,
+		FRready= p.FRready,
+		teamPrefix= teamprefix(p)
+	}
+end)
+
+rawset(_G, "FRIENDMOD_ToggleHUDstuff", function(b)
+	if (b==nil) then 
+		doHUDstuff= not doHUDstuff
+	else
+		doHUDstuff= b
+	end
+	return doHUDstuff
+end)
+--- ---
